ui/src/common/NetworkLostBanner.tsx

```diff
@@ -8,6 +8,14 @@ interface NetworkLostBannerProps {
 }
 
 export const NetworkLostBanner = ({height, retry}: NetworkLostBannerProps) => {
+    useEffect(() => {
```

Comment:
For state and logic we use mobx stores in our ui. I'd prefer having the reconnect logic there to.

It could be like this:
`CurrentUser.ts`
```typescript
private reconnectTimeoutId: number | null = null;

// other methods

public tryReconnect = (quiet = false) => {
    this.tryAuthenticate().catch(() => {
        if (!quiet) {
            this.snack('Reconnect failed');
        }
    });
};

private lostNetwork = () => {
    this.hasNetwork = false;
    if (this.reconnectTimeoutId !== null) {
        window.clearTimeout(this.reconnectTimeoutId);
    }
    this.reconnectTimeoutId = window.setInterval(() => this.tryReconnect(true), 3000);
};
```
Then replace `this.hasNetwork = false` with `this.lostNetwork()`.

Like this, logic stays inside the stores and our react components stay dumb.

In `Layout.tsx`, the `doReconnect`-method and the `reconnecting`-property should be deleted. And the `CurrentUser#tryReconnect` method should be forwarded to the `NetworkLostBanner`.

Reply:
Thanks for you hexaustive feedback!
I'm agree with the suggested logic: leave reconnect inside the stores and make dumb components.
I've made all the changes and I've added another `private` variable to `CurrentUser.ts`:

```ts
private readonly reconnectTime: number = 3000;
````

This constant (create with `readonly` ts keyword) could be useful to calibrate the amount of time of our timeout object.

Let me know if there are other suggestions!

Reply:
Looks pretty good! I only have one small remark left: https://github.com/gotify/server/pull/228#discussion_r334276230